### PR TITLE
Lightweight Shell

### PR DESCRIPTION
**Pull request by blitz**
@phip1611 mentioned that our shell environment is pretty heavy and can compile for quite some time when entered. This PR drops everything from the shell environment that is not immediately useful for compile the Rust code or style checking.

While I was here, I also applied `cargo machete` suggestions.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -259,8 +243,6 @@
               config.packages.stub
               config.packages.tool
             ];
-
-            TEST_SYSTEMD = pkgs.systemd;
```
Comment (nikstur) [inline flake.nix:263]: We should keep this around, otherwise you can't run the lzbt tests from the shell anymore.

**Review by nikstur — CHANGES_REQUESTED** [review]
Otherwise this looks good

**Review by RaitoBezarius — APPROVED** [review]
LGTM modulo what @nikstur said.
Some of the tools are useful for us, developers, e.g. OpenSSL, sbctl overidden, etc. But it's fine I guess.

File: flake.nix (hunk 1 above)
Comment (blitz) [reply flake.nix:263]: Good point. `cargo test` in lzbt passes now and I've also left openssl and sbctl.

**Comment by blitz** [comment]
@nikstur Can you check again?

**Pull request merged**